robot/basestation/static/js/roslibjs/ros-helpers.js

```diff
@@ -167,9 +187,32 @@ function initRosWeb () {
     messageType: 'geometry_msgs/Point'
   })
   battery_temps_listener.subscribe(function (message) {
-    $('#battery-temp-1').text(parseFloat(message.x).toFixed(2))
-    $('#battery-temp-2').text(parseFloat(message.y).toFixed(2))
-    $('#battery-temp-3').text(parseFloat(message.z).toFixed(2))
+    let t1 = parseFloat(message.x).toFixed(2)
+    let t2 = parseFloat(message.y).toFixed(2)
+    let t3 = parseFloat(message.z).toFixed(2)
+
+    $('#battery-temp-1').text(t1)
+    $('#battery-temp-2').text(t2)
+    $('#battery-temp-3').text(t3)
+
+    let temperatures = [t1, t2, t3]
+
+    for (i = 1; i <= temperatures.length ; i++) {
+      let temp = temperatures[i-1]
+      let color = 'white'
+
+      if (temp > MAX_TEMP || temp < MIN_TEMP) {
+        color = 'red'
+        errorSound()
+        if (temp > MAX_TEMP) {
+          navModalMessage('Warning: Battery temperature too high', 'Decreace temperature')
+        } else if (temp < MIN_TEMP) {
+          navModalMessage('Warning: Battery temperature too low', 'Increace temperature')
+        }
+      }
+
+      $('#battery-temp-' + i).css({'color': color})
```

Comment:
Ah I see why you used a for loop now. Then you should not iterate the values but the jquery objects so you won't even have to do that